packages/product-catalogue/src/components/ViewDetails/index.tsx

```diff
@@ -0,0 +1,76 @@
+import { ProductCatalogue } from '../../ducks/productCatalogue';
+import React from 'react';
+import { Col, Typography, Space } from 'antd';
+import { Resource404 } from '@opensrp/react-utils';
+import { Dictionary } from '@onaio/utils';
+
+const { Text } = Typography;
+
+/**
+ * helper function that creates a list of key,value pairs.
+ * This pairs are iterated and displayed as they are.
+ *
+ * @param {ProductCatalogue} product - a single product object
+ * @returns {Dictionary} - the mapping inform of an array
+ */
+export const defaultExtractViewDetails = (product: ProductCatalogue) => {
+  const mapping: Dictionary = {};
+  mapping['Product name'] = product.productName;
```

Comment:
How do you feel about having magic strings here?

Reply:
These still fall under the translate-ability question?, this has not yet been resolved ,and is pending both [here](https://github.com/OpenSRP/web/issues/82) and [here](https://github.com/OpenSRP/web/pull/41).

Reply:
Sorry I thought that these actually get rendered but I might be mistaken

Reply:
yeah, no they do get rendered. I will move them to string constants.